cpp/src/io/orc/stripe_data.cu

```diff
@@ -1579,12 +1579,25 @@ __global__ void __launch_bounds__(block_size)
           s->top.data.max_vals       = min(s->top.data.max_vals, blockDim.x);
         }
         __syncthreads();
-        n = numvals - ((s->top.data.max_vals + 7) >> 3);
-        if (t < n) {
-          secondary_val = s->vals.u8[((s->top.data.max_vals + 7) >> 3) + t];
-          if (t == 0) { s->top.data.buffered_count = n; }
+        if (s->top.data.max_vals >= s->top.data.index.run_pos[CI_DATA]) {
```

Comment:
To me it looks like this can be refactored into something like
```
auto const is_last_set = s->top.data.max_vals < s->top.data.index.run_pos[CI_DATA];
auto const xyz = is_last_set ? (blockDim.x >> 3) : ((s->top.data.max_vals + 7) >> 3); // don't know what this is, but it's used in multiple places
auto const next_numvals = is_last_set ? blockDim.x : min(numvals << 3u, s->top.data.max_vals);
n = numvals - xyz;
if (t < n) {
  secondary_val = s->vals.u8[xyz + t];
  if (t == 0) { s->top.data.buffered_count = n; }
}
numval = next_numval;
```